modules/solid_mechanics/neml2/src/LAROMANCE6DInterpolation.cxx

```diff
@@ -0,0 +1,385 @@
+{
+  Real factor = param[0];
+  Real lowerbound = param[1];
+  Real upperbound = param[2];
+  Real logmin = param[3];
+  Real logmax = param[4];
+  Real range = upperbound - lowerbound;
+  auto transformed_data =
+      (math::pow(10.0, ((data - lowerbound) * (logmax - logmin) / range) + logmin) - factor);
+  return transformed_data;
+}
+
+Scalar
+LAROMANCE6DInterpolation::transform_min_max(const Scalar & data,
+                                                   const std::vector<Real> & param) const
+{
+  Real data_min = param[0];
+  Real data_max = param[1];
+  Real scaled_min = param[2];
+  Real scaled_max = param[3];
+  auto transformed_data =
+      ((data - data_min) / (data_max - data_min)) * (scaled_max - scaled_min) + scaled_min;
+  return transformed_data;
+}
+
+std::string
+LAROMANCE6DInterpolation::json_to_string(std::string key)
+{
+  if (!_json.contains(key))
+    throw NEMLException("The key '" + std::string(key) + "' is missing from the JSON data file.");
+
+  std::string name = _json[key].template get<std::string>();
+  return name;
+}
+
+std::vector<Real>
+LAROMANCE6DInterpolation::json_to_vector(std::string key)
+{
+  if (!_json.contains(key))
+    throw NEMLException("The key '" + std::string(key) + "' is missing from the JSON data file.");
+
+  std::vector<Real> data_vec = _json[key].template get<std::vector<Real>>();
+  return data_vec;
+}
+
+Scalar
+LAROMANCE6DInterpolation::json_vector_to_torch(std::string key)
+{
+  if (!_json.contains(key))
+    throw NEMLException("The key '" + std::string(key) + "' is missing from the JSON data file.");
+
+  std::vector<Real> in_data = _json[key].template get<std::vector<Real>>();
+  const long long data_dim = in_data.size();
+  return Scalar(torch::from_blob(in_data.data(), {data_dim}).clone());
+}
+
+Scalar
+LAROMANCE6DInterpolation::json_6Dvector_to_torch(std::string key)
+{
+  if (!_json.contains(key))
+    throw NEMLException("The key '" + std::string(key) + "' is missing from the JSON data file.");
+
+  std::vector<std::vector<std::vector<std::vector<std::vector<std::vector<Real>>>>>> out_data =
+      _json[key]
+          .template get<
+              std::vector<std::vector<std::vector<std::vector<std::vector<std::vector<Real>>>>>>>();
+
+  std::vector<Real> linearize_values;
+  for (auto && level0 : out_data)
+    for (auto && level1 : level0)
+      for (auto && level2 : level1)
+        for (auto && level3 : level2)
+          for (auto && level4 : level3)
+            for (auto && value : level4)
+              linearize_values.push_back(value);
+
+  long long sz_l0 = out_data.size();
+  long long sz_l1 = out_data[0].size();
+  long long sz_l2 = out_data[0][0].size();
+  long long sz_l3 = out_data[0][0][0].size();
+  long long sz_l4 = out_data[0][0][0][0].size();
+  long long sz_l5 = out_data[0][0][0][0][0].size();
+  return Scalar(
+      torch::from_blob(linearize_values.data(), {sz_l0, sz_l1, sz_l2, sz_l3, sz_l4, sz_l5})
+          .clone());
```

Comment:
I'm surprised this doesn't utterly fail for you. `from_blob` requires a pointer to a contiguous block of values. You are setting this up as nested vectors, which contain a control block and have data allocated on the heap. There is _zero_ guarantee that this results in contiguous memory.